R/summarize.R

```diff
@@ -1,29 +1,85 @@
 #' @export
-#' @importFrom dplyr summarise_
+#' @importFrom dplyr bind_cols summarise_ select
+#' @importFrom purrr map_chr map_df
 #' @importFrom crunch crtabs
 summarise_.CrunchDataset <- function (.data, ..., .dots) {
     dots <- all_dots(.dots, ..., all_named = TRUE)
-    fmla <- dots_to_formula(dots, groups(.data))
-    out <- as_tibble(crtabs(fmla, data=.data))
-    ## Sort the return columns based on the request order
-    out <- out[, c(setdiff(names(out), names(dots)), names(dots))]
+    unweighted <- dots %>% map_chr(~as.character(.$expr)[[1]]) == "unweighted_n"
+    counts <- dots[unweighted]
```

Comment:
I found `counts` and `non_counts` hard to follow in the code below. I had to keep scrolling up to see what they really meant, which is `unweighted_n_measures` and `measures`, or something like that.